readthedocs/core/unresolver.py

```diff
@@ -141,46 +170,61 @@ def unresolve(self, url, append_indexhtml=True):
         :param append_indexhtml: If `True` directories will be normalized
          to end with ``/index.html``.
         """
-        parsed = urlparse(url)
-        domain = self.get_domain_from_host(parsed.netloc)
+        parsed_url = urlparse(url)
+        domain = self.get_domain_from_host(parsed_url.netloc)
         unresolved_domain = self.unresolve_domain(domain)
+        return self._unresolve(
```

Comment:
What's the difference between `def unresolve` and `def _unresolve`? It would be good to use more descriptive names for them because it's hard to know what each of them does by reading their names.

Reply:
there is the docstring, not sure about other names for it.

Reply:
Following the nomenclature you've been following with the other `unresolve_*` methods, I'd say that the current `unresolve` should be called `unresolve_url`. So, we will have:

- `unresolve_url`
- `unresolve_path`
- `unresolve_domain`
- `unresolve_domain_from_request`
- `_unresolve`

Also, `_unresolve_path` should be called `unresolve_path_with_parent_project`.

I want to avoid having methods that are called exactly the same, with the only difference that one has a `_` prefix. That's super confusing when following/reading the code. I think the names I'm proposing a lot more descriptive and easy to follow.

I'm 👍🏼 moving forward with these two changes to the names.

Maybe, also, `unresolve_path` could be renamed to `unresolve_path_from_unresolved_domain`, 😄